src/tui.rs

```diff
@@ -403,6 +403,15 @@ impl ToipeTui {
         //   - won't hang there waiting for user to type space
         lines.push(Text::from(line.join(" ")).with_faint());
 
+        if lines.len() >= terminal_height.into() {
+            return Err(ToipeError {
+                msg: format!(
+                    "toipe requires atleast {} lines in your terminal",
+                    lines.len()
+                ),
+            });
```

Comment:
This is good, but can be improved a bit. You could `impl From<&str> for ToipeError` (or perhaps `impl From<String> for ToipeError` to avoid references) and construct a `ToipeError` inside it taking a string.

Then this part will reduce to:
```rust
return Err(ToipeError::from(format!("toipe requires atleast {} lines in your terminal", lines.len())))
```

Reply:
implemented `From<String>` for `ToipeError`. and refactored code.